### PR TITLE
Only add architecture option if not a Win64 Generator

### PR DESCRIPTION
**Pull request by jpfeuffer**

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Should fix #95 
One win machine had a git update which removed it from the path. IT needs to fix it.
We test it here with -G "Visual Studio X Y" -A x64. So it should be confirmed by someone else.
I tested it quickly with -G "Visual Studio X Y Win64" though and it worked now.
When manually running the libraries after our Script configured them, they
reported "Generator ... Win64 does not support platform specification". So it seems like
you cannot specify both (which we did for the subcalls).

**Comment by jpfeuffer** [comment]
@hroest Maybe you can test it. Everyone I asked, has VS 2019. Which you probably also should upgrade to, when you find the time since otherwise OpenMP is broken.

**Comment by hroest** [comment]
> @hroest Maybe you can test it. Everyone I asked, has VS 2019. Which you probably also should upgrade to, when you find the time since otherwise OpenMP is broken.

we cannot since we need to build pyOpenMS which requires 14.x https://wiki.python.org/moin/WindowsCompilers -- but we do compile without OpenMP

**Comment by jpfeuffer** [comment]
VS 2019 is 14.2. Since 2015 the runtime is unified and backwards compatible.

File: CMakeLists.txt [hunk 1]
```diff
@@ -367,7 +367,9 @@ if (MSVC)
   else()
     set(BOOST_ARG "address-model=64")
     set(WIN_PLATFORM_ARG "x64")
-    set(ARCHITECTURE_OPTION_CMAKE "-A" "x64")
+    if (NOT CMAKE_GENERATOR MATCHES ".*Win64.*")
```
Comment (hroest) [inline CMakeLists.txt]:
```suggestion
    # Fix bug #95
    if (NOT CMAKE_GENERATOR MATCHES ".*Win64.*")
```

**Comment by hroest** [comment]
> VS 2019 is 14.2. Since 2015 the runtime is unified and backwards compatible.

ok, I did not know that! I guess I will try it out, thanks. 

> @hroest Maybe you can test it

testing it right now, it seems to work for HDF5, but I will also run for all contrib packages

**Comment by jpfeuffer** [comment]
Great thank you! let me know. Then I will accept your change and merge.

**Comment by hroest** [comment]
yes, it works now on my machine. Feel free to merge

**Pull request merged**